### DISCUSSION TITLE
Page loading performance slow

### DISCUSSION THREAD

**Discussion by samg666** [Budibase/budibase#4253] [Help]
Hello,

I have a Order page loading very slow, below is the screenshot of browser performance record, it seems the Scripting take over 11s. I am not sure whether it's due to my self-hosted server performance issue, or MySQL server performance issue, or something else. The button click script is quite simple as below, and my page has two `Data Provider / Query` . Appreciate if anyone can provide some guidence on how to find out where is the bottleneck.

Button Click > Nativage To > JavaScript
```
let url = "/orders/summary/"
let pram = new URLSearchParams();
pram.set("OrderStatus", "Pending") ;
pram.set("Keyword", $("Form1.Fields.Keywords"));
return url + encodeURI(pram);
```

<img width="2518" alt="iShot2022-01-29 21 39 00" src="https://user-images.githubusercontent.com/29649651/151663219-d2688d31-6261-4439-8ad2-ae1b8e131d01.png">

**Comment by lanmower** [comment]
my budibase is slow af, only thing holding us back from using it is the terrible speed.

**Reply by melohagan** [reply]
Could you provide your setup and some context please? I may be able to investigate your performance issues further.

**Reply by Keda87** [reply]
@melohagan we had a similar issue that the page loading is very slow and chrome profiling says the same that the scripting is the most time-consuming. I saw that the API calls in the network section are done in milliseconds but the page loaded about 15-17seconds

**Reply by melohagan** [reply]
Thanks @Keda87 
It may not be related, but it could be worth keeping an eye on this issue as well: https://github.com/Budibase/budibase/issues/6580

**Reply by Keda87** [reply]
haven't checked my server usage yet, will get back to you and thank you for pointing it out @melohagan

**Reply by samg666** [reply]
@Keda87 You may also need to check if your users has the issue access to below URLs as Budibase needs to load some JS/fonts/icons files from them. I had raised another issue on #4421 

<img width="281" alt="image" src="https://user-images.githubusercontent.com/29649651/178300219-69cc53b6-9695-49c2-91e7-76f4f350093b.png">

**Comment by v3ss0n** [comment]
![image](https://user-images.githubusercontent.com/419606/184142576-8ea9cbd6-90df-4b79-86ba-de29ef4df747.png)

What is happening here? Why loading budibase-client.js need 20s just to show , especially in modals.

Anyway to cache it? my system resources are seem to be okay.

Is the containers memory / cpu limited ? Beacue i still have 20GB of free ram.

**Reply by melohagan** [reply]
Hey @v3ss0n are you using many nested repeaters or anything like that?

**Reply by shogunpurple** [reply]
@v3ss0n - how are you running budibase? Is this cloud/self hosted, what does your hardware look like? if budibase-client.js is taking a long time to show, it's likely to be network related

**Reply by chaoticefx** [reply]
@shogunpurple I am also seeing that budibase-client.js?v=1.2.47 is taking 15-25s to load. I'm self-hosting it on an internal LAN server using the docker-compose linked here https://docs.budibase.com/docs/docker-compose and talking to the page directly so network issues should be minimized.

I don't have any nested repeaters, or any data sources at all. My testing page currently only contains either 1 embedded map or 1 embedded iframe, both of which exhibit the same slow initial load. Once budibase-client.js is loaded everything else seems to load relatively quickly and refreshing it is fast.

**Reply by v3ss0n** [reply]
Sorry for the late reply , i was busy with a project and no notification comes in from github . 
@melohagan it was  without nesting  , 2 tables . 3 datasource ,  1 for users , one for tasks , 1 for backlog. its quite small (only 200 rows , none of them joined ). 
Its not CPU problem , CPU is not even using much at all , it seems transfer rate of JS files are  so slow. 

@shogunpurple 
I am running it on  :

```
           .-/+oossssoo+/-.               
        `:+ssssssssssssssssss+:`           ------------- 
      -+ssssssssssssssssssyyssss+-         OS: Ubuntu 20.04.4 LTS x86_64 
    .ossssssssssssssssssdMMMNysssso.       Host: SYS-5039MC-H12TRF 0123456789 
   /ssssssssssshdmmNNmmyNMMMMhssssss/      Kernel: 5.4.0-89-generic 
  +ssssssssshmydMMMMMMMNddddyssssssss+     Uptime: 217 days, 17 hours, 1 min 
 /sssssssshNMMMyhhyyyyhmNMMMNhssssssss/    Packages: 1350 (dpkg), 13 (snap) 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Shell: bash 5.0.17 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Terminal: etterminal 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   CPU: Intel Xeon E-2276G (12) @ 4.900GHz 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   GPU: 05:00.0 ASPEED Technology, Inc. ASPEED Graphics Family 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Memory: 24206MiB / 64225MiB 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.
 /sssssssshNMMMyhhyyyyhdNMMMNhssssssss/                            
  +sssssssssdmydMMMMMMMMddddyssssssss+                             
   /ssssssssssshdmNNNNmyNMMMMhssssss/
    .ossssssssssssssssssdMMMNysssso.
      -+sssssssssssssssssyyyssss+-
        `:+ssssssssssssssssss+:`
            .-/+oossssoo+/-.
```

**Reply by melohagan** [reply]
Thanks for getting back! Are you still seeing the issue today? Have you tried the usual of updating to the latest version of Budibase and restarting the containers?

**Reply by v3ss0n** [reply]
> @v3ss0n - how are you running budibase? Is this cloud/self hosted, what does your hardware look like? if budibase-client.js is taking a long time to show, it's likely to be network related

Network is fine , what solve the problem is daily reboot of containers.

**Comment by ashik-entym** [comment]
Having similar issues with the published app on self hosted container. Sometimes taking more than 60 seconds to lead. The builder on the other seems better.

**Reply by melohagan** [reply]
Hey @ashik-entym 

What version of Budibase are you running, and how much data are you loading in your app. What does the component tree look like - is there a lot of repeaters / nesting?

**Reply by ashik-entym** [reply]
@melohagan  Just tried upgrading to 2.5. Still seeing these issues. 
![image](https://user-images.githubusercontent.com/102288635/230540784-5ecbdc86-f2d6-443e-9b72-a90ae6918649.png)

I'm not loading a lot of data, just have a few tables which has like 20-30 rows that too paginated (server). One page has a  single nesting but that's it. Still exploring budibase.

**Comment by cantaspinar** [comment]
I have a similar problem. When I send a REST API request in the Data console, Time is around 70 ms. However, the preview response is around 350-400ms. This is just the data page btw. I wonder why there is a huge difference.

<img width="830" alt="ss" src="https://github.com/Budibase/budibase/assets/14237377/a163bd0b-29de-4e54-852b-ca17d04bdb68">
